vulnerabilities/msgvalue-loop.md

```diff
@@ -0,0 +1,44 @@
+# Using ``msg.value`` in a loop.
+
+The value of ``msg.value`` in a transaction’s call never gets updated, even if the called contract ends up sending some or all of the ETH to another contract. This means that using ``msg.value`` in ``for`` or ``while`` loops, without extra accounting logic, will either lead to the transaction reverting (when there are no longer sufficient funds for later iterations), or to the contract being drained (when the contract itself has an ETH balance).
+
+```solidity
+contract depositer {
+    function deposit(address weth) payable external {
+        for (uint i = 0; i < 5; i ++) {
+            WETH(weth).deposit{value: msg.value}();
+        }
+    }
+}
+```
+In the above example, first iteration will use all the ``msg.value`` sent and other iterations can:
+1. Drain the contract if some ETH balance exists inside the contract.
+2. Revert on zero value transfer if ETH balance doesn't exist inside the contract.
+3. Succeed with zero value transfer.
```

Comment:
Presumably this is only the case if we don't check the success of the external call?

Reply:
Maybe good to clarify if so

Reply:
Rather than checking success of the external call, I think it depends more on the implementation of the external call. If the external implementation doesn't check for zero value transfer, all iterations will succeed. 

Please review the latest fix!